### PR TITLE
Security Fixes

### PR DESCRIPTION
**Pull request by BastLeblanc**
*Issue #, if available:*

*Description of changes:*
Security fixes after a https://github.com/nccgroup/ScoutSuite audit (plus other security scans)
- S3 bucket access logging
- S3 bucket encryption
- drop_invalid_header_fields on ELB

And some other small fixes
- typo in stage.sh
- added a virtualenv disable flag for stage.sh. If stage.sh is used inside a virtualenv already
- 404 links in README.md homepage

*Description of testing performed to validate your changes (required if pull request includes CloudFormation or source code changes):*
- manual Cloudformation deployment
- integration tests on the deployed env
- some manual regression test on the UI

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by BastLeblanc** [comment]
automated test in progress https://github.com/aws-samples/retail-demo-store/actions/runs/3290669111/jobs/5423817210

**Comment by BastLeblanc** [comment]
- integrations test : ran OK on my instance and automated tests
- cfn-lint: ok (left the warning on unused variables or dependsOn redondant)

File: aws/cloudformation-templates/swagger-ui-pipeline.yaml [hunk 1]
```diff
@@ -194,7 +208,7 @@ Resources:
       BucketName: !Ref ArtifactBucket
 
   CodeBuildProject:
-    Description: Retail Demo Store CodeBuild project for Swagger UI
+    #Description: Retail Demo Store CodeBuild project for Swagger UI
```
Comment (james-jory) [inline aws/cloudformation-templates/swagger-ui-pipeline.yaml:211]: ??

File: aws/cloudformation-templates/amazonpay.yaml [hunk 2]
```diff
@@ -20,10 +20,6 @@ Parameters:
     Type: String
     Default: none
 
-  AmazonPayStoreId:
-    Type: String
-    Default: none
-
```
Comment (james-jory) [inline aws/cloudformation-templates/amazonpay.yaml:26]: Should this param be deleted from the [EE templates](https://github.com/aws-samples/retail-demo-store/tree/master/aws/cloudformation-templates/event-engine) too?

**Review by james-jory — COMMENTED** [review]
A couple small comments.

File: aws/cloudformation-templates/amazonpay.yaml (hunk 2 above)
Comment (BastLeblanc) [reply aws/cloudformation-templates/amazonpay.yaml:26]: It's still used in web-ui-pipeline.yaml for an env variable ... so I would think we still need it ?

https://github.com/aws-samples/retail-demo-store/blob/9b3324e4354acbc6b1a4c0f573fe97eb1bdb4dae/aws/cloudformation-templates/web-ui-pipeline.yaml#L404

**Review by james-jory — APPROVED** [review]

**Pull request merged**